dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,235 @@
+        }
+
+    }
+
+    class MyTailerThread extends Thread {
+
+        Tailer tailer;
+        EventOutput eventOutput;
+        MyTailerListener listener;
+        String fileName;
+
+        @Override
+        public final void run() {
+
+            final OutboundEvent.Builder eventBuilder = new OutboundEvent.Builder();
+            try {
+                int logNumber = 0;
+                int count = 1;
+                int pageNumber = 1;
+                while (!eventOutput.isClosed()) {
+                    final String write = listener.getOut(true).toString();
+                    if (write != null && write.length() > 0) {
+                        final String prepWrite = String.format(
+                                "<p class=\"log page%d\" data-page=\"%d\" data-logNumber=\"%d\" style=\"margin:0\">%s</p>",
+                                pageNumber, pageNumber, logNumber, write);
+
+                        eventBuilder.name("success");
+                        eventBuilder.data(Map.class,
+                                map("lines", prepWrite, "pageId", pageNumber));
+                        eventBuilder.mediaType(MediaType.APPLICATION_JSON_TYPE);
+                        final OutboundEvent event = eventBuilder.build();
+                        eventOutput.write(event);
+
+                        count++;
+                        logNumber++;
+                        if (count % (LINES_PER_PAGE + 1) == 0) {
+                            pageNumber++;
+                            count = 1;
+                        }
+                    }
+                    Thread.sleep(1000);
+                }
+            } catch (Exception ex) {
+                Logger.warn(this.getClass(), "Stopping listening log events for " + fileName + ". Reason: " + ex.getMessage());
+            } finally {
+                stopTailer();
+                CloseUtils.closeQuietly(eventOutput);
+            }
+        }
+
+        public MyTailerThread(final File logFile, final MyTailerListener myTailerListener, final EventOutput myEventOutput, final int linesBack) {
+            final int linesNumber = linesBack <= 0? 5000: linesBack;
+            final long startPosition = ((logFile.length() - linesNumber) < 0) ? 0 : logFile.length() - linesNumber;
+            fileName = logFile.getName();
+            listener = myTailerListener;
+            listener.handle("Tailing info from file " + fileName);
+            listener.handle("----------------------------- ");
+            tailer = Tailer.create(logFile, listener, 1000);
+            tailer.setStartPosition(startPosition);
+            eventOutput = myEventOutput;
+        }
+
+
+        public void stopTailer() {
+            if (tailer != null) {
+                tailer.stop();
+            }
+        }
+
+        public void setTailer(Tailer tailer) {
+            this.tailer = tailer;
+        }
+
+        public EventOutput getEventOutput(){
+            return eventOutput;
+        }
+
+    }
```

Comment:
I don't know if we have to much logic in this end point, maybe we should move some into a API or Helper

Reply:
yeah probably This is refactor from a the old TaiLogServelt class

Reply:
but for now I think it's ok